tests/drivers/spi/spi_loopback/src/spi.c

```diff
@@ -137,15 +140,22 @@ static int spi_complete_loop(struct device *dev, struct spi_config *spi_conf)
 		return -1;
 	}
 
-	LOG_INF("Passed");
+	LOG_INF("Passed complete loop");
 
 	return 0;
 }
 
 
 static int spi_null_tx_buf(struct device *dev, struct spi_config *spi_conf)
 {
+#ifdef CONFIG_SPI_STM32_DMA
+/* if buffer address is null, the dma won't transfer */
```

Comment:
You'll need to find a way to follow the API, which is transferring 0s if tx buffer is NULL. Have a static tx buffer in your driver meant for that use-case, something like that.

Reply:
this is now handled in the `drivers/spi/spi_ll_stm32.c`, directly in the `spi_stm32_dma_tx_load` function : 
-->   /* if tx buff is null, then sends NOP on the line. */